### PR TITLE
feat: add creditCardIssuer

### PR DESCRIPTION
**Pull request by beninsydney**
Adds method for randomly selecting a credit card provider from the keys of `faker.definitions.finance.credit_card`

    > faker.finance.creditCardProvider()
    'visa'
    > faker.finance.creditCardProvider()
    'diners_club'

For pull request #882

### CODE REVIEW

File: src/finance.ts [hunk 1]
```diff
@@ -286,6 +286,19 @@ export class Finance {
     return cvv;
   }
 
+  /**
+   * Returns a random credit card provider
```
Comment (ST-DDT) [inline src/finance.ts:290]:
```suggestion
   * Returns a random credit card provider.
```

**Review by ST-DDT — CHANGES_REQUESTED** [review]
Please add a test for this.

````ts
expect(Object.keys(.....credit_card)).toContain(actual)
````

**Review by ST-DDT — DISMISSED** [review]

**Comment by Shinigami92** [comment]
Should we focus and merge https://github.com/faker-js/faker/pull/503 first?

File: src/finance.ts [hunk 2]
```diff
@@ -286,6 +286,19 @@ export class Finance {
     return cvv;
   }
 
+  /**
+   * Returns a random credit card provider.
+   *
+   * @example
+   * faker.finance.creditCardProvider() // 'discover'
+   */
+  creditCardProvider(): string {
```
[the remark this answers is not in the document]
Comment (ST-DDT) [reply src/finance.ts:295]: @pkuczynski Should we change the name of the parameter above too?

https://github.com/beninsydney/faker/blob/f0b7d871206c22090932c0e593eb5252eeb1efc3/src/finance.ts#L256

File: test/finance.spec.ts [hunk 3]
```diff
@@ -450,6 +450,16 @@ describe('finance', () => {
         });
       });
 
+      describe('cardCardProvider()', () => {
```
Comment (Shinigami92) [inline test/finance.spec.ts:453]:
```suggestion
      describe('cardCardIssuer()', () => {
```

File: test/finance.spec.ts [hunk 4]
```diff
@@ -450,6 +450,16 @@ describe('finance', () => {
...
+        it('should return a string', () => {
+          const provider = faker.finance.creditCardIssuer();
+          expect(provider).toBeTypeOf('string');
+          expect(Object.keys(faker.definitions.finance.credit_card)).toContain(
+            provider
```
Comment (Shinigami92) [inline test/finance.spec.ts:458]:
```suggestion
          const issuer = faker.finance.creditCardIssuer();
          expect(issuer).toBeTypeOf('string');
          expect(Object.keys(faker.definitions.finance.credit_card)).toContain(
            issuer
```

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

**Comment by beninsydney** [comment]
I've got this ready locally but I've noticed 'provider' is also used in `src/finance.ts` in `creditCardNumber()` and in `src/definitions/finance.ts` in a description block, do you want me to go ahead and update the nomenclature for consistency?

```
  /**
   * Generates a random credit card number.
   *
   * @param provider The name of the provider (case insensitive) or the format used to generate one.
   *
   * @example
   * faker.finance.creditCardNumber() // '4427163488668'
   * faker.finance.creditCardNumber('visa') // '4882664999003'
   * faker.finance.creditCardNumber('63[7-9]#-####-####-###L') // '6375-3265-4676-6644'
   */
  creditCardNumber(provider = ''): string {
    let format: string;
    const localeFormat = this.faker.definitions.finance.credit_card;
    const normalizedProvider = provider.toLowerCase();
    if (normalizedProvider in localeFormat) {
      format = this.faker.random.arrayElement(localeFormat[normalizedProvider]);
    } else if (provider.match(/#/)) {
      // The user chose an optional scheme
      format = provider;
    } else {
      // Choose a random provider
      // Credit cards are in an object structure
      const formats = this.faker.random.objectElement(localeFormat, 'value'); // There could be multiple formats
      format = this.faker.random.arrayElement(formats);
    }
    format = format.replace(/\//g, '');
    return this.faker.helpers.replaceCreditCardSymbols(format);
  }
```

File: src/finance.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (pkuczynski) [reply src/finance.ts]: Yes, I would say so...

**Comment by Shinigami92** [comment]
test/finance.spec.ts line 397

**Review by Shinigami92 — DISMISSED** [review]

**Review by ST-DDT — DISMISSED** [review]

File: test/finance.spec.ts [hunk 5]
```diff
@@ -450,6 +450,16 @@ describe('finance', () => {
         });
       });
 
+      describe('cardCardIssuer()', () => {
```
Comment (pkuczynski) [inline test/finance.spec.ts:453]:
```suggestion
      describe('creditCardIssuer()', () => {
```

**Review by pkuczynski — DISMISSED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — DISMISSED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**